shopinvader_schema_invoice/schemas/invoice.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2023 Camptocamp SA
+# @author: Simone Orsi <simahawk@gmail.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from datetime import date
+
+from extendable_pydantic import StrictExtendableBaseModel
+
+from .amount import InvoiceAmount
+
+
+class Invoice(StrictExtendableBaseModel, extra="ignore"):
+    id: int
+    name: str
+    date_invoice: date
+    date_due: date | None = None
+    state: str
+    payment_state: str
+    ref: str | str
+    payment_reference: str | str
```

Comment:
Why not 

```suggestion
    ref: str | None
    payment_reference: str | None
```

Reply:
As this is normally I string I'd rather stick to an empty string as default.
Do you see any implications?

Reply:
It may be a matter of test. `str | None` perhaps communicates more explicitly that the value can be set or not set.

Reply:
`str | None` means the value can be a string or None. I don't understant what `str | str` means. If you want an empty string by default it should be declared as 'ref : str = ""'